System Tests in GitHub CI template aren't removed when skipped

### Steps to reproduce

```shell
rails new blog --skip-system-test
```

Push up code to GitHub repository and run CI tests.

### Expected behavior

System tests are skipped and should not be run in CI.

### Actual behavior

System tests are being included in the template and encounter error when trying to be ran.

```
/opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundled_gems.rb:74:in `require': cannot load such file -- /home/runner/work/blog/blog/test/system (LoadError)
	from /opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/zeitwerk-2.7.1/lib/zeitwerk/core_ext/kernel.rb:34:in `require'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/test_unit/runner.rb:62:in `block in load_tests'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/test_unit/runner.rb:60:in `each'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/test_unit/runner.rb:60:in `load_tests'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/test_unit/runner.rb:52:in `run'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/commands/test/test_command.rb:33:in `perform'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/commands/test/test_command.rb:63:in `system'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/thor-1.3.2/lib/thor/command.rb:28:in `run'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/thor-1.3.2/lib/thor/invocation.rb:127:in `invoke_command'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/command/base.rb:178:in `invoke_command'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/thor-1.3.2/lib/thor.rb:538:in `dispatch'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/command/base.rb:73:in `perform'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/command.rb:65:in `block in invoke'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/command.rb:143:in `with_argv'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/command.rb:63:in `invoke'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/railties-8.0.0/lib/rails/commands.rb:18:in `<main>'
	from /opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /home/runner/work/blog/blog/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from bin/rails:4:in `<main>'
Error: Process completed with exit code 1.
```

### System configuration
**Rails version**: Rails 8.0.0
**Ruby version**: ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [arm64-darwin23]

--- a/railties/lib/rails/generators/rails/app/templates/github/ci.yml.tt
+++ b/railties/lib/rails/generators/rails/app/templates/github/ci.yml.tt
@@ -126,12 +126,12 @@ jobs:
           DATABASE_URL: postgres://postgres:postgres@localhost:5432
           <%- end -%>
           # REDIS_URL: redis://localhost:6379/0
-        <%- if options[:api] -%>
+        <%- if options[:api] || options[:skip_system_test] -%>
         run: bin/rails db:test:prepare test
         <%- else -%>
         run: bin/rails db:test:prepare test test:system
         <%- end -%>
-      <%- unless options[:api] -%>
+      <%- unless options[:api] || options[:skip_system_test] -%>
 
       - name: Keep screenshots from failed system tests
         uses: actions/upload-artifact@v4

--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
@@ -544,7 +544,9 @@ class AppGeneratorTest < Rails::Generators::TestCase
     assert_no_directory("test/system")
 
     assert_file ".github/workflows/ci.yml" do |content|
-      assert_no_match(/google-chrome-stable/, content)
+      assert_match(/db:test:prepare test/, content)
+      assert_no_match(/test:system/, content)
+      assert_no_match(/screenshots/, content)
     end
   end
 

